### PR TITLE
Modify rule S4999,S5000: LaYC format

### PR DESCRIPTION
**Pull request by marco-antognini-sonarsource**
## Review

A dedicated reviewer checked the rule description successfully for:

- [x] logical errors and incorrect information
- [x] information gaps and missing content
- [x] text style and tone
- [x] PR summary and labels follow [the guidelines](https://github.com/SonarSource/rspec/#to-modify-an-existing-rule)

### CODE REVIEW

File: rules/S5000/cfamily/rule.adoc [hunk 1]
```diff
@@ -83,8 +85,13 @@ bool isSame(Resource *r1, Resource *r2)
...
 == Resources
 
-https://en.cppreference.com/w/cpp/language/classes#Trivially_copyable_class[Definition of a trivially copyable class].
+=== Documentation
+
+* {cpp} reference - https://en.cppreference.com/w/cpp/language/classes#Trivially_copyable_class[Definition of a trivially copyable class].
+
+=== Related rules
 
+* S4999 - "memcpy", "memmove", and "memset" should only be called with pointers to trivially copyable types
```
Comment (alejandro-alvarez-sonarsource) [inline rules/S5000/cfamily/rule.adoc:94]:
```suggestion
* S4999 - `memcpy`, `memmove`, and `memset` should only be called with pointers to trivially copyable types
```

File: shared_content/cfamily/trivially_copyable_types.adoc [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+// Ignoring cv-qualified versions of these types here for brevity.
+
+A class is trivially copyable when:
+
+ * all its non-static data members and base classes are trivially copyable types,
+ * it has no virtual functions or base classes,
+ * its destructor is trival,
+ * and one or more of the following special member functions is trivial, and the rest are deleted: copy constructor, move constructor, copy assignment operator, and move assignment operator.
+
+_Note: a default implementation, either explicit (with `= default`) or implicit (if the special member function is omitted), is considered trivial._
```
Comment (alejandro-alvarez-sonarsource) [inline shared_content/cfamily/trivially_copyable_types.adoc:15]:
```suggestion
_Note: a default implementation is always considered trivial, both when it is explicit (with `= default`) or implicit (if the special member function is omitted)._
```

**Review by alejandro-alvarez-sonarsource — CHANGES_REQUESTED** [review]

File: rules/S4999/cfamily/rule.adoc [hunk 3]
```diff
@@ -1,54 +1,76 @@
...
 
+== Resources
+
+=== Documentation
+
+* {cpp} reference - https://en.cppreference.com/w/cpp/language/classes#Trivially_copyable_class[Definition of a trivially copyable class].
+
+=== Related rules
 
+* S5000 - "memcmp" should only be called with pointers to trivially copyable types with no padding
```
[the remark this answers is not in the document]
Comment (marco-antognini-sonarsource) [reply rules/S4999/cfamily/rule.adoc:73]: I agree it's not great but the [policy](https://github.com/SonarSource/rspec/blob/master/docs/link_formatting.adoc) is to use the title verbatim, so with regular quotes `"`.

**Review by marco-antognini-sonarsource — COMMENTED** [review]
Thanks for the nice suggestions!

**Review by alejandro-alvarez-sonarsource — APPROVED** [review]

**Pull request merged**